contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,519 @@
+                _fillQuote(_swaps[i]);
+                uint256 componentBalanceAfter = IERC20(components[i]).balanceOf(address(this));
+                componentAmountSold = componentBalanceBefore.sub(componentBalanceAfter);
+                require(maxAmountSell >= componentAmountSold, "ExchangeIssuance: OVERSOLD COMPONENT");
+            }
+
+        }
+        uint256 outputTokenBalanceAfter = _outputToken.balanceOf(address(this));
+        totalOutputTokenBought = totalOutputTokenBought.add(outputTokenBalanceAfter.sub(outputTokenBalanceBefore));
+    }
+
+    /**
+     * Execute a 0x Swap quote
+     *
+     * @param _quote          Swap quote as returned by 0x API
+     *
+     */
+    function _fillQuote(
+        bytes memory _quote
+    )
+        internal
+        
+    {
+
+        (bool success, bytes memory returndata) = swapTarget.call(_quote);
+
+        // Forwarding errors including new custom errors
+        // Taken from: https://ethereum.stackexchange.com/a/111187/73805
+        if (!success) {
+            if (returndata.length == 0) revert();
+            assembly {
+                revert(add(32, returndata), mload(returndata))
+            }
+        }
+
+    }
+
+    /**
+     * Transfers given amount of set token from the sender and redeems it for underlying components.
+     * Obtained component tokens are sent to this contract. 
+     *
+     * @param _setToken     Address of the SetToken to be redeemed
+     * @param _amount       Amount of SetToken to be redeemed
+     */
+    function _redeemExactSet(ISetToken _setToken, uint256 _amount, address _issuanceModule) internal returns (uint256) {
+        _setToken.safeTransferFrom(msg.sender, address(this), _amount);
+        IBasicIssuanceModule(_issuanceModule).redeem(_setToken, _amount, address(this));
+    }
+
+    /**
+     * Returns excess input token
+     *
+     * @param _inputToken         Address of the input token to return
+     * @param _receivedAmount     Amount received by the caller
+     * @param _spentAmount        Amount spent for issuance
+     */
+    function _returnExcessInputToken(IERC20 _inputToken, uint256 _receivedAmount, uint256 _spentAmount) internal {
+        uint256 amountTokenReturn = _receivedAmount.sub(_spentAmount);
+        if (amountTokenReturn > 0) {
+            _inputToken.safeTransfer(msg.sender,  amountTokenReturn);
+        }
+    }
+
+    /**
+     * Returns component positions required for issuance 
+     *
+     * @param _issuanceModule    Address of issuance Module to use 
+     * @param _setToken          Set token to issue
+     * @param _amountSetToken    Amount of set token to issue
+     */
+    function getRequiredIssuanceComponents(address _issuanceModule, ISetToken _setToken, uint256 _amountSetToken) public view returns(address[] memory components, uint256[] memory positions) {
+        try IDebtIssuanceModule(_issuanceModule).getRequiredComponentIssuanceUnits(_setToken, _amountSetToken) 
```

Comment:
As suggested I removed the issuance module whitelist. 
Since we previously used the issuance module registration to keep track of wether it is a Basic or Debt-Issuancemodule, I switched to just using this try / catch block, which seems to work but might not be optimal. (For example it will "swallow" an revert reason that might be thrown in  `getRequiredComponentIssuanceUnits` and it is probably also not gas efficient).

@ncitron Let me know if you have an idea for a better way of determining wether a given issuance module is basic or debt issuance. (Of course we can also bring back a mapping in this contract but I thought the point was to avoid that).

Reply:
As @ncitron suggested I removed the try/catch logic and added an `_isDebtIssuance` flag to the redeem / issuance methods to let the user specify wether given issuance module is basic or debt issuance.